Packet++/header/GvcpLayer.h

```diff
@@ -0,0 +1,758 @@
+ */
+namespace pcpp
+{
+	uint16_t hostToNet16(uint16_t val);  // forward declare the function in SystemUtil.h
+	uint16_t netToHost16(uint16_t val);  // forward declare the function in SystemUtil.h
+
+	namespace internal
+	{
+		static constexpr size_t kGvcpMagicNumber = 0x42;
+		static constexpr size_t kGvcpRequestHeaderLength = 8;
+		static constexpr size_t kGvcpAckHeaderLength = 8;
+		static constexpr size_t kGvcpDiscoveryBodyLength = 248;
+		static constexpr size_t kGvcpForceIpBodyLength = 56;
+	}  // namespace internal
+
+	using GvcpFlag = uint8_t;  // flag bits are specified by each command
+
+	/// @brief Gvcp command defines the command values and the corresponding acknowledge values
+	/// See more in the spec "18 Command and Acknowledge Values"
+	enum class GvcpCommand : uint16_t
+	{
+		// Discovery Protocol Control
+		DiscoveredCmd = 0x0002,
+		DiscoveredAck = 0x0003,
+		ForceIpCmd = 0x0004,
+		ForceIpAck = 0x0005,
+
+		// Streaming Protocol Control
+		PacketResendCmd = 0x0040,
+		PacketResendAck = 0x0041,  // Resent packet must be on the stream channel
+
+		// Device Memory Access
+		ReadRegCmd = 0x0080,
+		ReadRegAck = 0x0081,
+		WriteRegCmd = 0x0082,
+		WriteRegAck = 0x0083,
+		ReadMemCmd = 0x0084,
+		ReadMemAck = 0x0085,
+		WriteMemCmd = 0x0086,
+		WriteMemAck = 0x0087,
+		PendingAck = 0x0089,
+
+		// Asynchronous Events
+		EventCmd = 0x00C0,
+		EventAck = 0x00C1,
+		EventDataCmd = 0x00C2,
+		EventDataAck = 0x00C3,
+
+		// Miscellaneous
+		ActionCmd = 0x0100,
+		ActionAck = 0x0101,
+		Unknown = 0xFFFF
+	};
+
+	/// output operator for GvcpCommand
+	std::ostream& operator<<(std::ostream& os, GvcpCommand command);
+
+	/// @brief GVCP response status can be returned in an acknowledge message or a GVSP header.
+	/// See more in the spec "Table 19-1: List of Standard Status Codes"
+	enum class GvcpResponseStatus : uint16_t
+	{
+		Success = 0x0000,         ///< Command executed successfully
+		PacketResend = 0x0100,    ///< Only applies to packet being resent
+		NotImplemented = 0x8001,  ///< Command is not supported by the device
+		InvalidParameter =
+		    0x8002,  ///< At least one parameter provided in the command is invalid (or out of range) for the device
+		InvalidAddress = 0x8003,  ///< An attempt was made to access a non-existent address space location.
+		WriteProtect = 0x8004,    ///< The addressed register cannot be written to
+		BadAlignment = 0x8005,    ///< A badly aligned address offset or data size was specified.
+		AccessDenied =
+		    0x8006,  ///< An attempt was made to access an address location which is currently/momentary not accessible
+		Busy = 0x8007,  ///< A required resource to service the request is not currently available. The request may be
+		                ///< retried at a later time
+		LocalProblem = 0x8008,       // deprecated
+		MsgMismatch = 0x8009,        // deprecated
+		InvalidProtocol = 0x800A,    // deprecated
+		NoMsg = 0x800B,              // deprecated
```

Comment:
These `// deprecated` comments are not in the doxygen format so these values won't appear in the API documentation. We should change them to `///< deprecated`.

Same for other `// deprecated` comments below

Reply:
fixed.